cura/PrinterOutput/NetworkedPrinterOutputDevice.py

```diff
@@ -213,7 +213,7 @@ def post(self, target: str, data: str, on_finished: Optional[Callable[[QNetworkR
         request = self._createEmptyRequest(target)
         self._last_request_time = time()
         if self._manager is not None:
-            reply = self._manager.post(request, data)
+            reply = self._manager.post(request, data.encode())
```

Comment:
Did you try to call this function?
I have tried to call data.encode() and it returns "AttributeError: 'bytes' object has no attribute 'encode'"
Also while testing I discovered Cura does not support UM3 old firmware (I tested 3.7...).

Reply:
On https://github.com/Ultimaker/Cura/pull/4678/files/517428a308fcff91c68c54a226ad95ec63ac81fd#diff-0bb654171ac2bf659b787a51c1f10823L211 data is defined as a string. If the legacy UM3 output device supplies a bytes array, it is doing it wrong.

This PR can be implemented in a different way too, and that is to redefine the attribute types of the post method. You tell me, what should I do?

Reply:
Note: I don't have a UM3 (with or without legacy firmware), so I cannot test with that. I saw that the typing and the implementation do not match, so I fixed that.

Reply:
Made a PR to fix this: https://github.com/fieldOfView/Cura/pull/4